File: rb/features.rb
# OpenskyNetwork SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module OpenskyNetworkFeatures
  def self.make_feature(name)
    case name
    when "base"
      OpenskyNetworkBaseFeature.new
    when "ratelimit"
      OpenskyNetworkRatelimitFeature.new
    when "retry"
      OpenskyNetworkRetryFeature.new
    when "test"
      OpenskyNetworkTestFeature.new
    when "timeout"
      OpenskyNetworkTimeoutFeature.new
    else
      OpenskyNetworkBaseFeature.new
    end
  end
end

File: py/features.py
# OpenskyNetwork SDK feature factory

from feature.base_feature import OpenskyNetworkBaseFeature
from feature.test_feature import OpenskyNetworkTestFeature


def _make_feature(name):
    features = {
        "base": lambda: OpenskyNetworkBaseFeature(),
        "test": lambda: OpenskyNetworkTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
